storage/queue.go

```diff
@@ -144,21 +164,28 @@ func (bq *baseQueue) Start(clock *hlc.Clock, stopper *stop.Stopper) {
 func (bq *baseQueue) MaybeAdd(rng *Range, now proto.Timestamp) {
 	bq.Lock()
 	defer bq.Unlock()
+	should, priority := bq.impl.shouldQueue(now, rng)
+	bq.addInternal(rng, should, priority)
+}
 
-	if bq.disabled {
-		return
+// addInternal adds the range the queue with specified priority. If the
```

Comment:
it might be helpful to enumerate the reasons a range would not be added here. Better yet, return an error instead of a boolean explaining why the rage was not added.